components/script/serviceworker_manager.rs

```diff
@@ -198,8 +199,8 @@ impl ServiceWorkerManager {
 
     fn receive_message(&mut self) -> Option<Message> {
         select! {
-            recv(self.own_port.select(), msg) => msg.map(Message::FromConstellation),
-            recv(self.resource_receiver.select(), msg) => msg.map(Message::FromResource),
+            recv(self.own_port) -> msg => msg.ok().map(Message::FromConstellation),
+            recv(self.resource_receiver) -> msg => msg.ok().map(Message::FromResource),
```

Comment:
Let's revert this whole method to (roughly) what it used to be:

```rust
fn receive_message(&mut self) -> Result<Message, RecvError> {
    select! {
        recv(self.own_port) -> msg => msg.map(Message::FromConstellation),
        recv(self.resource_receiver) -> msg => msg.map(Message::FromResource),
    }
}
```

There's a `Some` in `handle_message` that'll need to be changed to `Ok`.